daml-lf/engine/src/main/scala/com/digitalasset/daml/lf/engine/Blinding.scala

```diff
@@ -47,16 +49,13 @@ object Blinding {
         }
         .mkString(";")
     }
-    if (enrichedTx.failedAuthorizations.isEmpty) {
-      Right(
-        BlindingInfo(
-          disclosure = enrichedTx.explicitDisclosure,
-          divulgence = enrichedTx.implicitDisclosure,
-        ))
+    if (failedAuthorizations.isEmpty) {
```

Comment:
Just for your information. 
To build an either conditionally you could use `Either.cond` function.
Explicit use of `if ... then ... else ...` is also fine

Reply:
ok. thanks

on a very slightly related issue (it's to do with conditional code!)
I noticed a coding pattern which looks rather inefficient, and I wonder if we care.

In `AuthorizingTransaction.scala` (prev `EnrichedTransaction.scala` there are many calls to `authorize` of the form:
```
        .authorize(
          nodeId = nodeId,
          passIf = create.signatories subsetOf auth.authParties,
          failWith = FailedAuthorization.CreateMissingAuthorization(
            templateId = create.coinst.template,
            optLocation = create.optLocation,
            authorizingParties = auth.authParties,
            requiredParties = create.signatories,
          ),
        )
```
which construct a `FailedAuthorization` object, even when `passIf` is false.

Maybe this could be improved by using a non-strict/by-name argument to the `authorize` function.
Is this regarded as good scala style?

Reply:
I do not really care ;). 
In the case you are describing, call-by-name argument seem pretty reasonable.